lib/actions/device_ts.ts

```diff
@@ -19,6 +19,35 @@ import { stripIndent } from 'common-tags';
 import { normalizeUuidProp } from '../utils/normalization';
 import * as commandOptions from './command-options';
 
+export const supported: CommandDefinition<{}, {}> = {
+	signature: 'devices supported',
+	description: 'list all supported devices',
+	help: stripIndent`
+		Use this command to get the list of all supported devices.
+
+		Examples:
+
+			$ balena devices supported
+	`,
+	async action(_params, _options, done) {
+		const sdk = (await import('balena-sdk')).fromSharedOptions();
+		const visuals = await import('resin-cli-visuals');
+		const _ = await import('lodash');
+
+		return sdk.models.config
+			.getDeviceTypes()
+			.then(deviceTypes => {
+				const fields = ['slug', 'name'];
+				deviceTypes = _.sortBy(deviceTypes, fields).filter(
+					dt => dt.state !== 'DISCONTINUED',
+				);
+				return visuals.table.horizontal(deviceTypes, fields) as string;
+			})
+			.then(console.log)
+			.nodeify(done);
```

Comment:
Something I noticed, which I think was already the case in the CoffeeScript implementation: the code is both returning a promise **and** explicitly calling the `done()` callback, so maybe `done()` will be called twice, once here explicitly and once by the Capitano framework, when it detects that a promise was returned. Perhaps completely harmless... or perhaps there's some side effect (?). You could leave it as it is, or you could choose to call `done()` here and return `undefined` (or any non-promise value) (in which case, the action could not be declared `async`, because an `async` function always returns a promise), or you could return a promise and not call `done()` here (i.e. remove the `.nodeify(done)` line and rename the action argument to `_done`).

Also: if you chose to declare the action `async` (thus necessarily returning a promise), then I thought of mentioning a "current team preference" (not just CLI, several other modules / projects too) of using async/await instead of promise chains in the implementation of an `async` function. For example, it might have been possible to write the code above (I haven't tested...) with something like:
```
async action(_params, _options, _done) {
    ...
    let deviceTypes = await sdk.models.config.getDeviceTypes();
    const fields = ['slug', 'name'];
    deviceTypes = _.sortBy(deviceTypes, fields).filter(dt => dt.state !== 'DISCONTINUED');
    const horizontalTable = await visuals.table.horizontal(deviceTypes, fields);
    console.log(horizontalTable);
```

Reply:
What I wrote above about `done()` being called twice is probably wrong (or conditionally correct...). When a promise is returned by the action, it looks like Capitano will only "automatically" call `done()` if `done` is not declared as an argument of the action function:

https://github.com/balena-io/capitano/blob/v1.9.0/lib/command.coffee#L84
`(_value) => callback() if @action.length < 3`

I.e., it should work if the action is declared with two arguments:
`async action(_params, _options)`
though I'm not sure if the TypeScript type checking would be happy with that.

Reply:
Ah yes, thanks.   I had straight translated from the coffeescript, and hadn't really analysed the logic well enough. I'll look into that (It seems this `.nodify(done)` pattern is used extensively in the coffeescript actions).

Thanks for the info on team preference re promises & async/await - very valuable!  I usually assume that the existing code is doing it the preferred was, and so follow that.  Please keep these kind of comments coming if you notice anything :)

Right, while we are translating we might as well change to the preferred way of doing things too, I'll change this to an `async` function.